fps-game-jam-org/open-game-jam-2020
# Feature/scene controller

Implemented scene controller / title controller and most of the game state controller with level intro and level end menu.

Tests:
1. Test Credits Page
  * Open and run the Title scene
  * Click on the "Credits" button
  * Verify the Credits scene is shown (Pass/ Fail)
  * Click on the "Back" button
  * Verify the Title scene is shown (Pass / Fail)


2. Test Playing the Game
2A. First Level Load, End Screen, and Replay Level
  * Open and run the Title scene
  * Click on the "Play" button
  * Verify the SampleScene is loaded in the Hierarchy and shown (Pass / Fail)
  * Verify the level number is displayed for the default 3 seconds. (Pass / Fail)
  * Verify the screen goes blank for 3 seconds (This is where the level would be played) (Pass / Fail)
  * Verify the level end state is shown (50/50 random for testing) (Pass / Fail)
  * Verify the level score is shown (Random float from 100 to 10000 for testing) (Pass / Fail)
  * Click on "Replay Level" ("Retry Level" if the level was lost)
  * Verify the same level is reloaded from the beginning and wait till the level end screen again (Pass / Fail)
  * Verify that the opposite level end state is shown (Repeat "Retry Level" if its the same as the first random outcome) (Pass / Fail)

2B. Next Level
  * Open and run the Title scene
  * Click on the "Play" button and wait for the level end screen
  * Keep replaying the level until it says "You Lose" on the end screen
  * Verify that the "Next Level" option is not shown on the end screen (Pass / Fail)
  * Keep replaying the level until is says "You Won" on the end screen
  * Verify that the "Next Level" option is shown on the end screen (Pass / Fail)
  * Click on "Next Level"
  * Verify the Title scene is shown since there is only one level (Pass / Fail)

2C. Exit to Title
  * Open and run the Title scene
  * Click on the "Play" button and wait for the level end screen
  * Verify "Exit" is shown on the level end screen
  * Click on "Exit"
  * Verify the Title scene is shown (Pass / Fail)


3. Test Game Exit
  * Build and run the game (Make sure to add the Title scene to the build settings)
  * Wait for the game to launch and click the "Exit" button
  * Verify the game application exits cleanly (Pass / Fail)
